VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.tsx

```diff
@@ -1,31 +1,56 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { useSelector } from 'react-redux'
 
 import { StackScreenProps } from '@react-navigation/stack'
 
 import { Button } from '@department-of-veterans-affairs/mobile-component-library'
 
+import { useDownloadLetter, useLetterBeneficiaryData } from 'api/letters'
+import { LetterTypeConstants, LettersDownloadParams } from 'api/types'
 import { AlertBox, BasicError, Box, FeatureLandingTemplate, LoadingComponent, TextArea, TextView } from 'components'
 import { NAMESPACE } from 'constants/namespaces'
 import { BenefitsStackParamList } from 'screens/BenefitsScreen/BenefitsStackScreens'
-import { RootState } from 'store'
-import { LetterTypeConstants } from 'store/api/types'
-import { LettersState, downloadLetter } from 'store/slices'
 import { testIdProps } from 'utils/accessibility'
 import { generateTestID } from 'utils/common'
-import { useAppDispatch, useTheme } from 'utils/hooks'
+import { useTheme } from 'utils/hooks'
 
 type GenericLetterProps = StackScreenProps<BenefitsStackParamList, 'GenericLetter'>
 
 function GenericLetter({ navigation, route }: GenericLetterProps) {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
-  const dispatch = useAppDispatch()
   const { header, description, letterType, descriptionA11yLabel } = route.params
-  const { downloading, letterDownloadError } = useSelector<RootState, LettersState>((state) => state.letters)
+  const [downloadLetter, setDownloadLetter] = useState(false)
+  const { data: letterBeneficiaryData } = useLetterBeneficiaryData()
+  const lettersOptions: LettersDownloadParams = {
+    militaryService: false,
+    serviceConnectedDisabilities: false,
+    serviceConnectedEvaluation: false,
+    nonServiceConnectedPension: letterBeneficiaryData?.benefitInformation.hasNonServiceConnectedPension || false,
+    monthlyAward: false,
+    unemployable: letterBeneficiaryData?.benefitInformation.hasIndividualUnemployabilityGranted || false,
+    specialMonthlyCompensation: letterBeneficiaryData?.benefitInformation.hasSpecialMonthlyCompensation || false,
+    adaptedHousing: letterBeneficiaryData?.benefitInformation.hasAdaptedHousing || false,
+    chapter35Eligibility: false,
+    deathResultOfDisability: letterBeneficiaryData?.benefitInformation.hasDeathResultOfDisability || false,
+    survivorsAward:
+      letterBeneficiaryData?.benefitInformation.hasSurvivorsIndemnityCompensationAward ||
+      letterBeneficiaryData?.benefitInformation.hasSurvivorsPensionAward ||
+      false,
+  }
+  const {
+    isLoading: downloading,
+    isError: letterDownloadError,
+    refetch: refetchLetter,
+  } = useDownloadLetter(letterType, lettersOptions, { enabled: downloadLetter })
```

Comment:
Same thing from my earlier comment applies here, since we don't want this query to run automatically, `enabled` can just be set to false in the hook, which removes the need for the `downloadLetter` state

Reply:
updated